src/module/scene/token-document/document.ts

```diff
@@ -29,12 +29,11 @@ class TokenDocumentPF2e<TParent extends ScenePF2e | null = ScenePF2e | null> ext
 
     /** Check actor for effects found in `CONFIG.specialStatusEffects` */
     override hasStatusEffect(statusId: string): boolean {
-        if (statusId === "dead") return this.overlayEffect === CONFIG.controlIcons.defeated;
-
         const actor = this.actor;
         if (!actor || !game.pf2e.settings.rbv) {
             return false;
         }
+        if (statusId === "dead") return actor.statuses.has("dead");
```

Comment:
This being after the rbv check seems odd, is that right?

Reply:
An actor is required for AEs so I moved it after the actor check but I think it makes no difference so it could stay at the top too. 
`if (statusId === "dead") return !!actor?.statuses.has("dead");` would lead to the same result I think.

Reply:
My concern is with || !game.pf2e.settings.rbv

Reply:
Oh right!🤦‍♂️
I'll move it to the top.